### ISSUE TITLE
kml is not gpx, can we have proper kml support

### ISSUE THREAD

**Issue by thorpie** [osmandapp/Osmand#5832]
kml is not gpx, can we have proper kml support

**Comment by black-square** [comment]
I wrote a script which can convert favourites and labels from `Google Maps` to `OsmAnd (GPX)` by using an Android device only:
https://github.com/black-square/gis_tools
Will it help with your use case?

**Comment by timguy** [comment]
For me this wouldn't be helpful as my source is really kml/kmz. 
And your described export from google is JSON as far as I understand googles description correct.
My source would be google as well but my maps - which can create a kmz link or download kml/kmz.

Best feature of course would be to just give a link to the kmz/kml URL and Osmand is tranforming it to gpx for the favourite import.

**Comment by axd1967** [comment]
(an interesting workaround is this: https://www.gpsvisualizer.com/map_input?form=googleearth)

**Comment by mnalis** [comment]
I think nowdays you can load `.kml` file directly in OsmAnd,so this issue is resolved?

Although https://www.osmand.net/docs/technical/osmand-file-formats/osmand-kml (Which contains workaround) should be updated to note KML support

**Comment by kleb007** [comment]
@mnalis you are right. For me it works since a longer time already to import KML (exported from My mapshttps://www.google.com/maps/d/) into OsmAnd.
So only documentation seems to be wrong.

Looks also like there is a class taking care of this since years (but maybe was merged later or was not used for import favorites)

https://github.com/osmandapp/OsmAnd/blame/ba209c48399b96b2fe5626784461de59e79698ca/OsmAnd/src/net/osmand/plus/helpers/Kml2Gpx.java#L16
